main.tf

```diff
@@ -333,3 +331,24 @@ resource "aws_vpn_gateway" "this" {
 
   tags = "${merge(var.tags, map("Name", format("%s", var.name)))}"
 }
+
+resource "aws_vpn_gateway_attachment" "vgw" {
+  count = "${var.attach_vpn_gateway != "default" ? 1 : 0}"
+
+  vpc_id         = "${aws_vpc.this.id}"
+  vpn_gateway_id = "${var.attach_vpn_gateway}"
+}
+
+resource "aws_vpn_gateway_route_propagation" "public" {
+  count = "${var.public_propagating_vgws && var.enable_vpn_gateway || var.public_propagating_vgws && var.attach_vpn_gateway != "default " ? length(var.public_subnets) : 0}"
+
+  route_table_id = "${element(aws_route_table.public.*.id, count.index)}"
+  vpn_gateway_id = "${element(concat(aws_vpn_gateway.this.*.id, aws_vpn_gateway_attachment.vgw.*.vpn_gateway_id), count.index)}"
+}
+
+resource "aws_vpn_gateway_route_propagation" "private" {
+  count = "${var.private_propagating_vgws && var.enable_vpn_gateway || var.private_propagating_vgws && var.attach_vpn_gateway != "default" ? length(var.private_subnets) : 0}"
```

Comment:
`count` can be updated like this:

`count = "${var.create_vpc && var.propagate_private_route_tables_vgw && (var.enable_vpn_gateway || var.vpn_gateway_id = "") ? length(var.private_subnets) : 0}"`